pkg/api/forge/v1alpha1/containerimagebuild_types.go

```diff
@@ -0,0 +1,226 @@
+type InitContainer struct {
+	// Name of the init container.
+	// +kubebuilder:validation:MinLength=1
+	Name string `json:"name"`
+
+	// Docker image name.
+	// +kubebuilder:validation:MinLength=1
+	Image string `json:"image"`
+
+	// Entrypoint array. The Docker image's ENTRYPOINT is used if this is not provided.
+	// +kubebuilder:validation:Optional
+	Command []string `json:"command"`
+
+	// Arguments to the entrypoint. The Docker image's CMD is used if this is not provided.
+	// +kubebuilder:validation:Optional
+	Args []string `json:"args"`
+
+	// Environment variables.
+	// +kubebuilder:validation:Optional
+	Env []corev1.EnvVar `json:"env"`
+}
+
+// ContainerImageBuildSpec defines the desired state of ContainerImageBuild
+type ContainerImageBuildSpec struct {
+	// Name used to build an image.
+	// +kubebuilder:validation:MinLength=1
+	ImageName string `json:"imageName"`
+
+	// Build context for the image. This can be a local path or url.
+	Context string `json:"context"`
+
+	// If the build context is a URL, the timeout in seconds for fetching. Defaults to 0, which disables the timeout.
+	// +kubebuilder:validation:Optional
+	ContextTimeoutSeconds uint16 `json:"contextTimeoutSeconds"`
+
+	// Push to one or more registries.
+	// +kubebuilder:validation:MinItems=1
+	PushRegistries []string `json:"pushTo"`
+
+	// Configure one or more registry hosts with special requirements.
+	// +kubebuilder:validation:Optional
+	Registries []Registry `json:"registries"`
+
+	// Image build arguments.
+	// +kubebuilder:validation:Optional
+	BuildArgs []string `json:"buildArgs"`
+
+	// Labels added to the image during build.
+	// +kubebuilder:validation:Optional
+	Labels map[string]string `json:"labels"`
+
+	// Limits build cpu consumption.
+	// +kubebuilder:validation:Optional
+	CPU string `json:"cpu"`
+
+	// Limits build memory consumption.
+	// +kubebuilder:validation:Optional
+	Memory string `json:"memory"`
+
+	// Resources are the requests and limits applied to image builds.
+	// +kubebuilder:validation:Optional
+	Resources corev1.ResourceRequirements `json:"resources,omitempty"`
+
+	// Optional deadline in seconds for image build to complete.
+	// +kubebuilder:validation:Optional
+	TimeoutSeconds uint16 `json:"timeoutSeconds"`
+
+	// Prevents images larger than this size (in bytes) from being pushed to a registry. By default,
+	// an image of any size will be pushed.
+	// +kubebuilder:validation:Optional
+	ImageSizeLimit uint64 `json:"imageSizeLimit"`
+
+	// Provide arbitrary data for use in plugins that extend default capabilities.
+	// +kubebuilder:validation:Optional
+	PluginData map[string]string `json:"pluginData"`
+
+	// Disable the use of layer caches during build.
+	// +kubebuilder:validation:Optional
+	DisableBuildCache bool `json:"disableBuildCache"`
+
+	// Disable export of layer cache when it is enabled.
+	// +kubebuilder:validation:Optional
+	DisableLayerCacheExport bool `json:"disableLayerCacheExport"`
```

Comment:
Do we need to track parity for a few of these? `DisableLayerCacheExport`, `DisableBuildCache`, `ImageSizeLimit`?

Reply:
How do `DisableLayerCacheExport` and `DisableBuildCache` work in V1? This model somewhat mimics that design with its localized disk cache so I'm not sure how to remove that but I can look.

I figured out how to implement `ImageSizeLimit` with https://github.com/dominodatalab/hephaestus/blob/67c91d24ea7e7f6ce64a4eb7cb3bf3e513157a12/pkg/buildkit/buildkit.go#L194-L195. It's just a question if we really want it. What do you think?

Reply:
I think `DisableLayerCacheExport` ( push cache to registry ) made sense in Forge but doesn't really have meaning in either V1 or V3. I think we do not need to worry about this, do you disagree?

`DisableBuildCache` can be implemented with the use of the `no-cache` attribute when performing a build. This will ensure a single build does not use the stored (filesystem) cache for any of its steps.

Reply:
> I think DisableLayerCacheExport ( push cache to registry ) made sense in Forge but doesn't really have meaning in either V1 or V3. I think we do not need to worry about this, do you disagree?

Is this maybe confusion over terminology? Are we not going to need [instruction caching at all](https://github.com/moby/buildkit/blob/master/README.md#inline-push-image-and-cache-together)? 

> DisableBuildCache can be implemented with the use of the no-cache attribute when performing a build. This will ensure a single build does not use the stored (filesystem) cache for any of its steps.

+1, but does the image build resource need a field for it?

Reply:
I've added these fields and I'm going to implement this support in another PR.